### ISSUE TITLE
BUG. New block may corrupt the grid

### ISSUE THREAD

**Issue by AndriiVel** [EPAM-Online-Courses/tetris#97]
When the grid is almost filled with blocks, a new block may appear on top (not above) of the already existing block. Moreover, the new block can "cut out" a part of already existing one.
Please refer to the images provided.

<details>
  <summary>Images</summary>

![1](https://user-images.githubusercontent.com/63800077/122580510-52cf1500-d056-11eb-9fcb-83d8665d8973.png)

![2](https://user-images.githubusercontent.com/63800077/122580518-55316f00-d056-11eb-8a40-0c0063ba55f3.png)

![3](https://user-images.githubusercontent.com/63800077/122580526-56fb3280-d056-11eb-99a7-9d74313f5c6f.png)

![4](https://user-images.githubusercontent.com/63800077/122580538-58c4f600-d056-11eb-8fb2-e81579fac0ff.png)

![5](https://user-images.githubusercontent.com/63800077/122580541-5a8eb980-d056-11eb-86ec-a1dcd915ae83.png)

![6](https://user-images.githubusercontent.com/63800077/122580547-5bbfe680-d056-11eb-89d1-fe1de47b41b8.png)

![7](https://user-images.githubusercontent.com/63800077/122580550-5d89aa00-d056-11eb-8599-8f8d6757eeee.png)

![8](https://user-images.githubusercontent.com/63800077/122580559-5f536d80-d056-11eb-8b25-7577ab5441f2.png)

![9](https://user-images.githubusercontent.com/63800077/122580563-611d3100-d056-11eb-81fc-3eb0ef53c2e6.png)

![10](https://user-images.githubusercontent.com/63800077/122580569-624e5e00-d056-11eb-8273-316d1510dde8.png)

![11](https://user-images.githubusercontent.com/63800077/122580585-64b0b800-d056-11eb-94c1-c16f0b42cbb5.png)
New block appears:
![12](https://user-images.githubusercontent.com/63800077/122580593-67131200-d056-11eb-8687-409c4871fc47.png)
And falls down:
![13](https://user-images.githubusercontent.com/63800077/122580669-7eea9600-d056-11eb-9673-4841983e8d15.png)

![14](https://user-images.githubusercontent.com/63800077/122580678-814cf000-d056-11eb-8d80-c9a094e87358.png)

and so on...
</details>

<details>
<summary>Last screen</summary>
The last screen printed by this example game is:

![last printed screen](https://user-images.githubusercontent.com/63800077/122580803-a3467280-d056-11eb-9380-0dfbc6c58fb3.png)

</details>

This is apparently not a feature but a bug, as it allows to score more than it normally would have been possible.

**Found in version:** 0.4
**Can be reproduced in versions:** 0.5

**Comment by KoLaR8** [comment]
Don't know if it's new issue or this one, but I noticed that this bug can lead to situation when some part of old block is levitating on grid, without any blok below. I pictured in on sceenshots: 
white "Y" block appears:
![obraz](https://user-images.githubusercontent.com/76537707/122735733-99ec1e80-d27f-11eb-8d90-5a406bcf54ff.png)
then This "I" block go through it:
![obraz](https://user-images.githubusercontent.com/76537707/122735951-cbfd8080-d27f-11eb-9497-cfe38ad303f8.png)
it lands:
![obraz](https://user-images.githubusercontent.com/76537707/122736238-1bdc4780-d280-11eb-8f34-60d2dc34d0f8.png)
and part of white "Y" block is still in the air :D

**Comment by AndriiVel** [comment]
I think it's the same. 
I guess one screen is missing, between your 1st and 2nd images.
The I block appears right on top of the Y block. By "on top" I mean not higher, but kind of "instead of". So the I block "corrupts" the grid, particularly the Y block which was already placed on the grid. 
So first it appears on top and "replaces"/"corrupts" some parts of the grid. Then it checks that there is nothing below it and it's free to fall down. Then it falls down.

Smth like this:
![image](https://user-images.githubusercontent.com/63800077/122741255-10d7e600-d285-11eb-8288-c0bfd81de7d6.png)

**Comment by KoLaR8** [comment]
Yes, I missed one screenshot. Thx for mentioning that. It's exactly how you describe it.
